src/Tizen.MachineLearning.Train/Tizen.MachineLearning.Train/Model.cs

```diff
@@ -0,0 +1,409 @@
+            Log.Info(NNTrainer.Tag, "Created Model");
+        }
+
+        /// <summary>
+        /// Constructs the neural network model with the given configuration file.
+        /// </summary>
+        /// <param name="modelConf">The nntrainer model configuration file.</param>
+        /// <since_tizen> 10 </since_tizen>
+        public Model(string modelConf)
+        {
+            if (string.IsNullOrEmpty(modelConf))
+                NNTrainer.CheckException(NNTrainerError.InvalidParameter, "modelConf is null");
+
+            NNTrainerError ret = Interop.Model.ConstructWithConf(modelConf, out handle);
+            NNTrainer.CheckException(ret, "Failed to create model instance with modelConf");
+            Log.Info(NNTrainer.Tag, "Created Model with Conf path: "+ modelConf);
+        }
+        /// <summary>
+        /// Destructor of Model
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        ~Model()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object.
+        /// </summary>
+        /// <since_tizen> 10 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object including opened handle.
+        /// </summary>
+        /// <param name="disposing">If true, disposes any disposable objects. If false, does not dispose disposable objects.</param>
+        /// <since_tizen> 10 </since_tizen>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposed)
+                return;
+            if (disposing)
+            {
+                // release managed object
+            }
+            // release unmanaged object
+            if (handle != IntPtr.Zero)
+            {
+                // Destroy the neural network model.
+                NNTrainerError ret = Interop.Model.Destroy(handle);
+                NNTrainer.CheckException(ret, "Failed to destroy model instance");
+
+                handle = IntPtr.Zero;
+            }
+            disposed = true;
+        }
+
+        /// <summary>
+        /// Compiles and finalizes the neural network model with the hyperparameter.
+        /// </summary>
+        /// <remarks>
+        /// Use this function to initialize neural network model.Various
+        /// hyperparameter before compile the model can be set. Once compiled,
+        /// any modification to the properties of model or layers/dataset/optimizer in
+        /// the model will be restricted. Further, addition of layers or changing the
+        /// optimizer/dataset of the model will not be permitted.
+        /// The input format of hyperparameter must be 'key = value' format.
+        /// <param name="hyperparameter">Hyperparameters for train complie.</param>
```

Comment:
typo, and for array the name of the parameter as plural would be better

Reply:
thank you.
hyperparameter is one or more.